### DISCUSSION TITLE
How can I post PDF or Image to Rest API?

### DISCUSSION THREAD

**Discussion by Kanary159357** [Budibase/budibase#6263] [Help] [answered]
I'm working on upload files to specific URL with REST API. Something like image, pdf etc....

I know that there are attachment component to upload files. But don't know how to post it via REST API. 

How can I use attachment files to upload using REST API?

**Comment by melohagan** [comment, marked as answer]
Hi @Kanary159357 

I don't know of a way to POST the content of a file attachment, but you can access the URL which could be posted to a service that is capable of grabbing a file from a URL.

Alternatively it may be possible to create an Automation with a bash script that can read and POST files from the MinIO storage, but that's beyond me.

**Reply by Kanary159357** [reply]
Well I was talking about
set file Object in request body and set header ```Content-Type:{fileType}``` and then call rest api with POST methods...

**Comment by realtothstar** [comment]
I have the same question, I upload a file over the attachment object, after that I need to transfer the file direct to nifi for post processing. How is that possible?

**Comment by HeerlijkeHeer** [comment]
I have the same question. I have a REST API endpoint that accepts the (binary) content of a file. I'd like to be able to have an Attachment form component where the user can drop his file and have its content automatically be sent to my REST API.
